pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py

```diff
@@ -78,76 +73,82 @@ def training_step(self, batch, batch_idx):
         }
 
     def validation_epoch_end(self, outputs):
-        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
+        avg_loss = torch.stack([x["val_loss"] for x in outputs]).mean()
 
-        tensorboard_logs = {'mse_loss': avg_loss}
+        tensorboard_logs = {"mse_loss": avg_loss}
 
-        return {
-            'val_loss': avg_loss,
-            'log': tensorboard_logs
-        }
+        return {"val_loss": avg_loss, "log": tensorboard_logs}
 
     def test_step(self, batch, batch_idx):
         loss = self._run_step(batch)
 
         return {
-            'test_loss': loss,
+            "test_loss": loss,
         }
 
     def test_epoch_end(self, outputs):
-        avg_loss = torch.stack([x['test_loss'] for x in outputs]).mean()
+        avg_loss = torch.stack([x["test_loss"] for x in outputs]).mean()
 
-        tensorboard_logs = {'mse_loss': avg_loss}
+        tensorboard_logs = {"mse_loss": avg_loss}
 
-        return {
-            'test_loss': avg_loss,
-            'log': tensorboard_logs
-        }
+        return {"test_loss": avg_loss, "log": tensorboard_logs}
 
     def configure_optimizers(self):
         return torch.optim.Adam(self.parameters(), lr=self.hparams.learning_rate)
 
     @staticmethod
     def add_model_specific_args(parent_parser):
         parser = ArgumentParser(parents=[parent_parser], add_help=False)
-        parser.add_argument('--hidden_dim', type=int, default=128,
-                            help='itermediate layers dimension before embedding for default encoder/decoder')
-        parser.add_argument('--latent_dim', type=int, default=32,
-                            help='dimension of latent variables z')
-        parser.add_argument('--input_width', type=int, default=28,
-                            help='input image width - 28 for MNIST (must be even)')
-        parser.add_argument('--input_height', type=int, default=28,
-                            help='input image height - 28 for MNIST (must be even)')
-        parser.add_argument('--batch_size', type=int, default=32)
-        parser.add_argument('--num_workers', type=int, default=8, help="num dataloader workers")
-        parser.add_argument('--learning_rate', type=float, default=1e-3)
-        parser.add_argument('--data_dir', type=str, default='')
+        parser.add_argument(
+            "--hidden_dim",
+            type=int,
+            default=128,
+            help="itermediate layers dimension before embedding for default encoder/decoder",
+        )
+        parser.add_argument("--latent_dim", type=int, default=32, help="dimension of latent variables z")
+        parser.add_argument("--learning_rate", type=float, default=1e-3)
         return parser
 
 
-def cli_main():
+def cli_main(args=None):
+    # cli_main()
     parser = ArgumentParser()
+    parser.add_argument("--dataset", default="mnist", type=str, help="mnist, cifar10, stl10, imagenet")
+    script_args, _ = parser.parse_known_args(args)
```

Comment:
what happen in `args=None`?

Reply:
It will parse CLI args 😄 ...both `parser.parse_args(None)` and `parser.parse_known_args(None)` behave this way. its the default value for that arg.